spec.md

```diff
@@ -3240,6 +3216,39 @@ Letters are converted to uppercase at the start of words, lowercase elsewhere.
 ```
 
 
+## Optional features
```

Comment:
Should Starlark have optional features? An alternative is to say Starlark permits recursion, and then anyone who wants to not implement it can document themselves as diverging from the spec.